Add support for threaded conversation

glorialimicrosoft opened the pull request:
# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

<a href="?expand=1&template=sdk_configuration_template.md">Click here</a> to open a PR for only SDK configuration.

glorialimicrosoft committed:
first pass on threaded conversation APIs

glorialimicrosoft committed:
use query for initiator and channel id, return 207 and error array for participant operations, fix some issues

glorialimicrosoft committed:
Update models.tsp

glorialimicrosoft committed:
cleanup

glorialimicrosoft committed:
some improvements

glorialimicrosoft committed:
Add ConversationMessageContent

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[CallAutomation-Messages](https://apiview.dev/Assemblies/Review/a78dc78241d64a12a78a8c9d1b9fb7b6?revisionId=2842c4f3703f4cd5b2abd27c487b729a)

glorialimicrosoft committed:
address api review feedback 1

rename initiator to participant,
fixing add/remove participants naming,
Use ResourceAction instead of ResourceCollectionAction

glorialimicrosoft committed:
address api review feedback 2

switching to ConversationMessageContent to support a broader range of message types.
Use ResourceAction (POST) with an action type of "terminate" instead of ResourceDelete

glorialimicrosoft committed:
use @withoutOmittedProperties decorator for ConversationMessageContent

glorialimicrosoft committed:
use ResourceCreateOrReplace for creating conversation instead of ResourceCollectionAction

glorialimicrosoft committed:
renaming and cleaning up

pohtsng commented on specification/communication/Communication.Messages/models.tsp:
```diff
@@ -450,6 +476,315 @@ model NotificationMessagesSendRequestBody {
+  topic?: string;
+
+  @doc("List of delivery channel IDs.")
+  deliveryChannelIds?: string[];
```
Is it possible for the business to own multiple Whatsapp/SMS/Email accounts and need to communicate with different external participants over different business accounts?

glorialimicrosoft committed:
make a few changes

/messages/conversations get    "name": "participant" => participantId
/messages/conversations/{conversationId}/messages get "name": "participant" => participantId
conversation model:  conversationId => Id
conversationMessageItem model: messageId => Id  visibility  none => read
Add a delete /messages/conversations/{conversationId} dor the real delete opeation
Use ResourceCreateWithServiceProvidedName for create converstion

glorialimicrosoft committed:
fix client error

glorialimicrosoft committed:
rewrite createConversation

have initial message as part of the request body
return conversation as a response body

glorialimicrosoft committed:
remove read only for participantIds in RemoveParticipantsRequest

glorialimicrosoft committed:
Address comments from Azure SDK Review

glorialimicrosoft committed:
fix naming

glorialimicrosoft committed:
fix naming body->options

glorialimicrosoft committed:
fix naming body->options for RemoveParticipantsOptions

glorialimicrosoft committed:
External -> external

jimchou-dev commented on specification/communication/Communication.Messages/models.tsp:
```diff
@@ -450,6 +476,336 @@ model NotificationMessagesSendRequestBody {
+  @doc("Details of a send conversation message request.")
+  request: ConversationMessageContent;
+
+  @doc("The options of the outbound delivery strategy for messages sent by emplyees in a conversation.\r\nSupports internalOnly, allChannels.")
```
change comment to participants

Commit:
Resolve merge conflict

Commit:
Rename version to 2025-04-01-preview

Commit:
Merge branch 'main' into gelli/threadedConversation

Commit:
fix lifecycle instead of string

jimchou-dev committed:
Merge branch 'main' into gelli/threadedConversation

Commit:
fix typo

Commit:
fix typo

glorialimicrosoft committed:
suppress and fix warnings

glorialimicrosoft committed:
add BearerAuth back and suppress the warning

glorialimicrosoft committed:
Merge branch 'main' into gelli/threadedConversation

glorialimicrosoft committed:
add version 2025-04-01-preview in the readme file.

glorialimicrosoft committed:
Merge branch 'gelli/threadedConversation' of https://github.com/Azure/azure-rest-api-specs into gelli/threadedConversation

glorialimicrosoft committed:
fix extra dash

glorialimicrosoft committed:
create examples

Commit:
add examples

Commit:
fix typos

jimchou-dev committed:
Merge branch 'main' into gelli/threadedConversation

Commit:
fix examples

Commit:
fix prettier

Commit:
fix examples

Commit:
fix examples

glorialimicrosoft committed:
fix breaking change

jimchou-dev committed:
Merge branch 'main' into gelli/threadedConversation

glorialimicrosoft committed:
fix breaking change warning

glorialimicrosoft committed:
Merge branch 'gelli/threadedConversation' of https://github.com/Azure/azure-rest-api-specs into gelli/threadedConversation

CHILIU-MSFT reviewed (approved):
LGTM

glorialimicrosoft committed:
Merge branch 'main' into gelli/threadedConversation
